### PR TITLE
[site] hardcode links on opentitan.org/documentation

### PR DESCRIPTION
**Pull request by hcallahan-lowrisc**
Due to the previous breakage of the site-builder container, at some point the hugo build broke such that it was not correctly resolving these links, and it was not noticed until the container issues were resolved (recently).
For now, hardcode the links. When the root-cause has been determined and fixed, revert this commit/PR.

### CODE REVIEW

**Review by marnovandermaas — APPROVED** [review]
Thanks for this fix. Let's do this to get a working site and then fix Hugo later.

**Comment by marnovandermaas** [comment]
I'll wait for CI to pass before merging

**Comment by marnovandermaas** [comment]
Merging this one now because the remaining CI tasks are about bitstreams and the site builder failing will be fixed in a follow up PR: https://github.com/lowRISC/opentitan/pull/18245
It's important that I merge this soon because it fixes links on the main documentation landing page.

**Comment by hcallahan-lowrisc** [comment]
@jwnrt FYI, I didn't give much details but essentially what was going wrong was the Hugo 'doc-url' shortcode was not being populated correctly, and the production site was not producing valid links.
e.g. `{{< doc-url >}}doc/project_governance{{< /doc-url >}}` would become -> `<no_value>/doc/project_governance`

I couldn't recreate this locally so I'm not sure what was going wrong. I don't think it's super high prio to fix, but if you have any ideas it would be great. I believe it broke in https://github.com/lowRISC/opentitan/pull/17996, but since the container-build broke at the same time it didn't become apparent until I fixed that in https://github.com/lowRISC/opentitan/pull/18245

**Comment by jwnrt** [comment]
@hcallahan-lowrisc Interesting, I can't tell what went wrong here from inspection. Passing in the URL as an environment variable that filled out a shortcode was a little hacky anyway so maybe there's a better way of doing this. Thanks for fixing in the meantime

**Pull request merged**
